src/libAtomVM/utils.h

```diff
@@ -259,4 +259,11 @@ static inline __attribute__((always_inline)) void *cast_func_to_void_ptr(func_pt
     #define NO_DISCARD(...)
 #endif
 
+#ifdef __GNUC__
```

Comment:
Do we use any compiler that doesn't define `__GNUC__`?
Also I'm not sure this should be considered a GNUC extension.

Both clang and gcc should support `__has_builtin(__builtin_unreachable)`

Reply:
I wouldn't rule out one day a VisualStudio build, it is something that I was looking for help for a long time (such as #8).

Reply:
Also I would `#define UNREACHABLE() unreachable()` if C23.

Reply:
For the records: `__has_builtin` cannot be used since it has been introduced in a newer GCC version.